dev/MRTCore/mrt/mrm/include/mrm/Results.h

```diff
@@ -0,0 +1,397 @@
+    //! \see DefStringResult_GetLength()
+    HRESULT GetLength(_Out_ size_t* length) const;
+
+    size_t GetLength() const;
+
+    //! \see DefStringResult_GetSize()
+    HRESULT GetSize(_Out_ size_t* size) const;
+
+    bool IsEmpty()
+    {
+        PCWSTR pRef = GetRef();
+        return ((pRef == nullptr) || (pRef[0] == L'\0'));
+    }
+
+    //! \see DefStringResult_Concat()
+    HRESULT Concat(_In_ PCWSTR str);
+
+    HRESULT ConcatPathElement(_In_ PCWSTR pathElement) { return ConcatPathElement(pathElement, L'\\'); }
+    HRESULT ConcatPathElement(_In_ PCWSTR pathElement, _In_ WCHAR separator);
+
+    HRESULT NormalizePathSlashes() { return NormalizePathSlashes(L'\\'); }
+    HRESULT NormalizePathSlashes(_In_ WCHAR wantSlash);
+
+    HRESULT IsAbsolutePath(_In_ WCHAR separator, _Out_ bool* absolutePath) const;
+    HRESULT IsAbsolutePath(_Out_ bool* absolutePath) const { return IsAbsolutePath(L'\\', absolutePath); }
+
+    bool IsAbsolutePath(_In_ WCHAR separator) const;
+    bool IsAbsolutePath() const { return IsAbsolutePath(L'\\'); }
+
+    HRESULT Truncate(_In_ size_t size);
+
+    HRESULT CompareWithOptions(_In_ PCWSTR str, _In_ DEFCOMPAREOPTIONS options, _Out_ DEFCOMPARISON* result) const;
+    HRESULT Compare(_In_ PCWSTR str, _Out_ DEFCOMPARISON* result) const;
+    HRESULT ICompare(_In_ PCWSTR str, _Out_ DEFCOMPARISON* result) const;
+    HRESULT CompareWithOptions(_In_ const StringResult* str, _In_ DEFCOMPAREOPTIONS options, _Out_ DEFCOMPARISON* result) const;
+    HRESULT Compare(_In_ const StringResult* str, _Out_ DEFCOMPARISON* result) const;
+    HRESULT ICompare(_In_ const StringResult* str, _Out_ DEFCOMPARISON* result) const;
+    DEFCOMPARISON CompareWithOptions(_In_ PCWSTR str, _In_ DEFCOMPAREOPTIONS options) const;
+    DEFCOMPARISON Compare(_In_ PCWSTR str) const;
+    DEFCOMPARISON ICompare(_In_ PCWSTR str) const;
+
+    void Inspect(void) const;
+
+    bool TryFindLastOf(_In_ WCHAR charToFind, _Out_ size_t* foundAtPosition) const;
+    bool TryFindFirstOf(_In_ WCHAR charToFind, _Out_ size_t* foundAtPosition) const;
+
+    HRESULT ReplaceAll(_In_ WCHAR charToFind, _In_ WCHAR charToReplaceWith, _Out_ size_t* charPositionOfFirstReplacement);
+
+    HRESULT SetCopyInteger(_In_ UINT32 value);
+
+    HRESULT CopyToOutParams(
+        _In_ size_t bufferSizeInChars,
+        _Out_writes_to_opt_(bufferSizeInChars, *numCharsWritten) PWSTR buffer,
+        _Out_opt_ size_t* numCharsWritten) const;
+
+    static HRESULT SetRef(_In_opt_ StringResult* str, _In_opt_ PCWSTR refString)
+    {
+        if (str != nullptr)
+        {
+            return str->SetRef(refString);
+        }
+        return E_INVALIDARG;
+    }
+
+    HRESULT Contains(_In_ PCWSTR str, _Out_ bool* result) const;
+
+    bool Contains(_In_ PCWSTR str) const;
+};
+
+class BlobResult : public DefObject
+{
+private:
+    DEFBLOBRESULT m_blobResult;
+
+public:
+    static HRESULT NewRef(
+        _In_reads_bytes_opt_(dataSizeInBytes) const void* dataToReference,
```

Comment:
(non-blocking_, we need to build a plan on what to do on incoming SAL code, which is not required (not supposed to be in code) per our coding standards.